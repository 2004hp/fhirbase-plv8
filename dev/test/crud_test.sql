--db:fhirb
--{{{
SELECT assert_eq(
  'base/a/b/c',
  _build_url('{"base":"base"}', 'a','b','c'),
  '_build_url'
);
SELECT assert_eq(
  'base/a/1/c',
  _build_url('{"base":"base"}', 'a',1::text,'c'),
  '_build_url'
);

SELECT assert_eq(
  '{"rel": "self", "href": "base/Patient/a9c6bcbe-3319-47dc-8e68-6cd318f79b94/_history/fba704fc-32f9-4f57-ae92-4c767c672b07"}'::jsonb,
  _build_link('{"base":"base"}', 'Patient','a9c6bcbe-3319-47dc-8e68-6cd318f79b94','fba704fc-32f9-4f57-ae92-4c767c672b07'),
  '_build_link'
);

SELECT assert_eq(
  'base/Patient/a9c6bcbe-3319-47dc-8e68-6cd318f79b94'::text,
  _build_id('{"base":"base"}', 'Patient','a9c6bcbe-3319-47dc-8e68-6cd318f79b94'),
  '_build_id'
);

SELECT assert_eq(
  'rid',
  _extract_id('rid/_history/vid'),
  '_extract_id'
);

SELECT assert_eq(
  'vid',
  _extract_vid('rid/_history/vid'),
  '_extract_vid'
);

--}}}
--{{{
\set cfg '{"base":"https://test.me"}'
\set pt `cat test/fixtures/pt.json`
\set pt2 `cat test/fixtures/pt2.json`
\set pt_tags '[{"scheme": "http://pt.com", "term": "http://pt/vip", "label":"pt"}]'

BEGIN;

WITH created AS (
  SELECT fhir_create(:'cfg'::jsonb, 'Patient', :'pt'::jsonb, :'pt_tags'::jsonb) bundle
)

SELECT assert_eq(c.bundle#>'{entry,0,content}', :'pt'::jsonb, 'fhir_create')
FROM created c;

ROLLBACK;
--}}}
