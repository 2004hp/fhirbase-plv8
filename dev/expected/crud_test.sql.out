SELECT assert_eq(
  'base/a/b/c',
  _build_url('{"base":"base"}', 'a','b','c'),
  '_build_url'
);
   assert_eq   
---------------
 OK _build_url
(1 row)

SELECT assert_eq(
  'base/a/1/c',
  _build_url('{"base":"base"}', 'a',1::text,'c'),
  '_build_url'
);
   assert_eq   
---------------
 OK _build_url
(1 row)

SELECT assert_eq(
  '{"rel": "self", "href": "base/Patient/a9c6bcbe-3319-47dc-8e68-6cd318f79b94/_history/fba704fc-32f9-4f57-ae92-4c767c672b07"}'::jsonb,
  _build_link('{"base":"base"}', 'Patient','a9c6bcbe-3319-47dc-8e68-6cd318f79b94','fba704fc-32f9-4f57-ae92-4c767c672b07'),
  '_build_link'
);
   assert_eq    
----------------
 OK _build_link
(1 row)

SELECT assert_eq(
  'base/Patient/a9c6bcbe-3319-47dc-8e68-6cd318f79b94'::text,
  _build_id('{"base":"base"}', 'Patient','a9c6bcbe-3319-47dc-8e68-6cd318f79b94'),
  '_build_id'
);
  assert_eq   
--------------
 OK _build_id
(1 row)

SELECT assert_eq(
  'rid',
  _extract_id('rid/_history/vid'),
  '_extract_id'
);
   assert_eq    
----------------
 OK _extract_id
(1 row)

SELECT assert_eq(
  'vid',
  _extract_vid('rid/_history/vid'),
  '_extract_vid'
);
    assert_eq    
-----------------
 OK _extract_vid
(1 row)

BEGIN;
BEGIN
WITH created AS (
  SELECT fhir_create('{"base":"https://test.me"}'::jsonb, 'Patient',  E'{
  "resourceType": "Patient",
  "text": {
    "status": "generated",
    "div": "<div>\\n      <p>\\n        <b>Generated Narrative</b>\\n      </p>\\n      <p>\\n        <b>identifier</b>: BSN = 123456789 (official), BSN = 123456789 (official)\\n      </p>\\n      <p>\\n        <b>name</b>: Roel(official)\\n      </p>\\n      <p>\\n        <b>telecom</b>: ph: +31612345678(mobile), ph: +31201234567(home)\\n      </p>\\n      <p>\\n        <b>gender</b>: \\n        <span title=\\"Codes: {http://snomed.info/sct 248153007}, {http://hl7.org/fhir/v3/AdministrativeGender M}\\">Male</span>\\n      </p>\\n      <p>\\n        <b>birthDate</b>: 13-Mar 1960\\n      </p>\\n      <p>\\n        <b>deceased[x]</b>: false\\n      </p>\\n      <p>\\n        <b>address</b>: Bos en Lommerplein 280 Amsterdam 1055RW NLD (home)\\n      </p>\\n      <p>\\n        <b>maritalStatus</b>: \\n        <span title=\\"Codes: {http://snomed.info/sct 36629006}, {http://hl7.org/fhir/v3/MaritalStatus M}\\">Legally married</span>\\n      </p>\\n      <p>\\n        <b>multipleBirth[x]</b>: false\\n      </p>\\n      <p>\\n        <b>photo</b>: \\n      </p>\\n      <h3>Contacts</h3>\\n      <table class=\\"grid\\">\\n        <tr>\\n          <td>\\n            <b>Relationship</b>\\n          </td>\\n          <td>\\n            <b>Name</b>\\n          </td>\\n          <td>\\n            <b>Telecom</b>\\n          </td>\\n          <td>\\n            <b>Address</b>\\n          </td>\\n          <td>\\n            <b>Gender</b>\\n          </td>\\n          <td>\\n            <b>Organization</b>\\n          </td>\\n        </tr>\\n        <tr>\\n          <td>\\n            <span title=\\"Codes: {http://snomed.info/sct 127850001}, {http://hl7.org/fhir/patient-contact-relationship partner}\\">Wife</span>\\n          </td>\\n          <td>Ariadne Bor-Jansma</td>\\n          <td>ph: +31201234567(home)</td>\\n          <td> </td>\\n          <td> </td>\\n          <td> </td>\\n        </tr>\\n      </table>\\n      <p>\\n        <b>communication</b>: \\n        <span title=\\"Codes: {urn:std:iso:639-1 nl-NL}\\">Dutch</span>\\n      </p>\\n      <p>\\n        <b>managingOrganization</b>: AUMC\\n      </p>\\n      <p>\\n        <b>active</b>: true\\n      </p>\\n    </div>"
  },
  "identifier": [
    {
      "use": "official",
      "label": "MRN",
      "system": "MRN",
      "value": "7777777"
    },
    {
      "use": "official",
      "label": "BSN",
      "system": "urn:oid:2.16.840.1.113883.2.4.6.3",
      "value": "123456789"
    }
  ],
  "name": [
    {
      "use": "official",
      "text": "Roel",
      "family": [
        "Bor"
      ],
      "given": [
        "Roelof Olaf"
      ],
      "prefix": [
        "Drs."
      ],
      "suffix": [
        "PDEng."
      ]
    }
  ],
  "telecom": [
    {
      "system": "phone",
      "value": "+31612345678",
      "use": "mobile"
    },
    {
      "system": "phone",
      "value": "+31201234567",
      "use": "home"
    }
  ],
  "gender": {
    "coding": [
      {
        "system": "http://snomed.info/sct",
        "code": "248153007",
        "display": "Male"
      },
      {
        "system": "http://hl7.org/fhir/v3/AdministrativeGender",
        "code": "M",
        "display": "Male"
      }
    ]
  },
  "birthDate": "1960-03-13",
  "deceasedBoolean": false,
  "address": [
    {
      "use": "home",
      "line": [
        "Bos en Lommerplein 280"
      ],
      "city": "Amsterdam",
      "zip": "1055RW",
      "country": "NLD"
    }
  ],
  "maritalStatus": {
    "coding": [
      {
        "system": "http://snomed.info/sct",
        "code": "36629006",
        "display": "Legally married"
      },
      {
        "system": "http://hl7.org/fhir/v3/MaritalStatus",
        "code": "M"
      }
    ]
  },
  "multipleBirthBoolean": false,
  "photo": [
    {
      "contentType": "image/jpeg",
      "url": "binary/@f006"
    }
  ],
  "contact": [
    {
      "relationship": [
        {
          "coding": [
            {
              "system": "http://snomed.info/sct",
              "code": "127850001",
              "display": "Wife"
            },
            {
              "system": "http://hl7.org/fhir/patient-contact-relationship",
              "code": "partner"
            }
          ]
        }
      ],
      "name": {
        "use": "usual",
        "text": "Ariadne Bor-Jansma"
      },
      "telecom": [
        {
          "system": "phone",
          "value": "+31201234567",
          "use": "home"
        }
      ]
    }
  ],
  "communication": [
    {
      "coding": [
        {
          "system": "urn:std:iso:639-1",
          "code": "nl-NL",
          "display": "Dutch"
        }
      ]
    }
  ],
  "managingOrganization": {
    "reference": "Organization/%s",
    "display": "AUMC"
  },
  "active": true
}'::jsonb, '[{"scheme": "http://pt.com", "term": "http://pt/vip", "label":"pt"}]'::jsonb) bundle
)
SELECT assert_eq(c.bundle#>'{entry,0,content}',  E'{
  "resourceType": "Patient",
  "text": {
    "status": "generated",
    "div": "<div>\\n      <p>\\n        <b>Generated Narrative</b>\\n      </p>\\n      <p>\\n        <b>identifier</b>: BSN = 123456789 (official), BSN = 123456789 (official)\\n      </p>\\n      <p>\\n        <b>name</b>: Roel(official)\\n      </p>\\n      <p>\\n        <b>telecom</b>: ph: +31612345678(mobile), ph: +31201234567(home)\\n      </p>\\n      <p>\\n        <b>gender</b>: \\n        <span title=\\"Codes: {http://snomed.info/sct 248153007}, {http://hl7.org/fhir/v3/AdministrativeGender M}\\">Male</span>\\n      </p>\\n      <p>\\n        <b>birthDate</b>: 13-Mar 1960\\n      </p>\\n      <p>\\n        <b>deceased[x]</b>: false\\n      </p>\\n      <p>\\n        <b>address</b>: Bos en Lommerplein 280 Amsterdam 1055RW NLD (home)\\n      </p>\\n      <p>\\n        <b>maritalStatus</b>: \\n        <span title=\\"Codes: {http://snomed.info/sct 36629006}, {http://hl7.org/fhir/v3/MaritalStatus M}\\">Legally married</span>\\n      </p>\\n      <p>\\n        <b>multipleBirth[x]</b>: false\\n      </p>\\n      <p>\\n        <b>photo</b>: \\n      </p>\\n      <h3>Contacts</h3>\\n      <table class=\\"grid\\">\\n        <tr>\\n          <td>\\n            <b>Relationship</b>\\n          </td>\\n          <td>\\n            <b>Name</b>\\n          </td>\\n          <td>\\n            <b>Telecom</b>\\n          </td>\\n          <td>\\n            <b>Address</b>\\n          </td>\\n          <td>\\n            <b>Gender</b>\\n          </td>\\n          <td>\\n            <b>Organization</b>\\n          </td>\\n        </tr>\\n        <tr>\\n          <td>\\n            <span title=\\"Codes: {http://snomed.info/sct 127850001}, {http://hl7.org/fhir/patient-contact-relationship partner}\\">Wife</span>\\n          </td>\\n          <td>Ariadne Bor-Jansma</td>\\n          <td>ph: +31201234567(home)</td>\\n          <td> </td>\\n          <td> </td>\\n          <td> </td>\\n        </tr>\\n      </table>\\n      <p>\\n        <b>communication</b>: \\n        <span title=\\"Codes: {urn:std:iso:639-1 nl-NL}\\">Dutch</span>\\n      </p>\\n      <p>\\n        <b>managingOrganization</b>: AUMC\\n      </p>\\n      <p>\\n        <b>active</b>: true\\n      </p>\\n    </div>"
  },
  "identifier": [
    {
      "use": "official",
      "label": "MRN",
      "system": "MRN",
      "value": "7777777"
    },
    {
      "use": "official",
      "label": "BSN",
      "system": "urn:oid:2.16.840.1.113883.2.4.6.3",
      "value": "123456789"
    }
  ],
  "name": [
    {
      "use": "official",
      "text": "Roel",
      "family": [
        "Bor"
      ],
      "given": [
        "Roelof Olaf"
      ],
      "prefix": [
        "Drs."
      ],
      "suffix": [
        "PDEng."
      ]
    }
  ],
  "telecom": [
    {
      "system": "phone",
      "value": "+31612345678",
      "use": "mobile"
    },
    {
      "system": "phone",
      "value": "+31201234567",
      "use": "home"
    }
  ],
  "gender": {
    "coding": [
      {
        "system": "http://snomed.info/sct",
        "code": "248153007",
        "display": "Male"
      },
      {
        "system": "http://hl7.org/fhir/v3/AdministrativeGender",
        "code": "M",
        "display": "Male"
      }
    ]
  },
  "birthDate": "1960-03-13",
  "deceasedBoolean": false,
  "address": [
    {
      "use": "home",
      "line": [
        "Bos en Lommerplein 280"
      ],
      "city": "Amsterdam",
      "zip": "1055RW",
      "country": "NLD"
    }
  ],
  "maritalStatus": {
    "coding": [
      {
        "system": "http://snomed.info/sct",
        "code": "36629006",
        "display": "Legally married"
      },
      {
        "system": "http://hl7.org/fhir/v3/MaritalStatus",
        "code": "M"
      }
    ]
  },
  "multipleBirthBoolean": false,
  "photo": [
    {
      "contentType": "image/jpeg",
      "url": "binary/@f006"
    }
  ],
  "contact": [
    {
      "relationship": [
        {
          "coding": [
            {
              "system": "http://snomed.info/sct",
              "code": "127850001",
              "display": "Wife"
            },
            {
              "system": "http://hl7.org/fhir/patient-contact-relationship",
              "code": "partner"
            }
          ]
        }
      ],
      "name": {
        "use": "usual",
        "text": "Ariadne Bor-Jansma"
      },
      "telecom": [
        {
          "system": "phone",
          "value": "+31201234567",
          "use": "home"
        }
      ]
    }
  ],
  "communication": [
    {
      "coding": [
        {
          "system": "urn:std:iso:639-1",
          "code": "nl-NL",
          "display": "Dutch"
        }
      ]
    }
  ],
  "managingOrganization": {
    "reference": "Organization/%s",
    "display": "AUMC"
  },
  "active": true
}'::jsonb, 'fhir_create')
FROM created c;
   assert_eq    
----------------
 OK fhir_create
(1 row)

ROLLBACK;
ROLLBACK
